tests/end-to-end/upgrade/pkg/tests/eventmesh-migration/helpers/event.go

```diff
@@ -0,0 +1,261 @@
+
+	subApis "github.com/kyma-project/kyma/components/event-bus/apis/eventing/v1alpha1"
+	ebClientSet "github.com/kyma-project/kyma/components/event-bus/client/generated/clientset/internalclientset"
+
+	"github.com/avast/retry-go"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/eventing/pkg/apis/eventing/v1alpha1"
+	eventingclientv1alpha1 "knative.dev/eventing/pkg/client/clientset/versioned/typed/eventing/v1alpha1"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+)
+
+const timeout = time.Second * 30
+
+func SendEvent(target, eventType, eventTypeVersion string) error {
+	log.Printf("Sending an event to target: %s with eventType: %s, eventTypeVersion: %s", target, eventType, eventTypeVersion)
+	payload := fmt.Sprintf(
+		`{"event-type":"%s","event-type-version": "%s","event-time":"2018-11-02T22:08:41+00:00","data":"foo"}`, eventType, eventTypeVersion)
+	transport := &http.Transport{
+		DialContext: (&net.Dialer{
+			Timeout: timeout,
+		}).DialContext,
+	}
+	client := http.Client{Transport: transport}
+	res, err := client.Post(target,
+		"application/json",
+		strings.NewReader(payload))
+	if err != nil {
+		return errors.Wrap(err, "HTTP POST request failed in SendEvent() ")
+	}
+
+	if err := verifyStatusCode(res, 200); err != nil {
+		return errors.Wrap(err, "HTTP POST request returned non-2xx failed in SendEvent() ")
+	}
+
+	return nil
+}
+
+func CheckEvent(target, eventType, eventTypeVersion string, retryOptions ...retry.Option) error {
+	return retry.Do(func() error {
+		transport := &http.Transport{
+			DialContext: (&net.Dialer{
+				Timeout: timeout,
+			}).DialContext,
+		}
+		client := http.Client{Transport: transport}
+		res, err := client.Get(target)
+		if err != nil {
+			return errors.Wrap(err, "HTTP GET failed in CheckEvent()")
+		}
+
+		if err := verifyStatusCode(res, 200); err != nil {
+			return errors.Wrap(err, "HTTP GET request returned non-2xx in CheckEvent()")
+		}
+		body, err := ioutil.ReadAll(res.Body)
+		defer res.Body.Close()
+
+		if err != nil {
+			errors.Wrap(err, "failed ReadAll() in CheckEvent")
+		}
+		var resp http.Header
+		err = json.Unmarshal(body, &resp)
+		if err != nil {
+			errors.Wrap(err, "failed Unmarshal() in CheckEvent")
+		}
+
+		err = res.Body.Close()
+		if err != nil {
+			errors.Wrap(err, "failed Close() CheckEvent")
+		}
+
+		if resp.Get("ce-type") != eventType {
+			return fmt.Errorf("event type wrong: Got %s, Wanted: %s", resp.Get("ce-type"), eventType)
+		}
+		if resp.Get("ce-eventtypeversion") != eventTypeVersion {
+			return fmt.Errorf("event type version wrong: Got %s, Wanted: %s", resp.Get("ce-eventtypeversion"), eventTypeVersion)
+		}
+
+		return nil
+	}, retryOptions...)
+
+	return nil
+}
+
+type TriggerOption func(trigger *v1alpha1.Trigger)
+
+func WithURISubscriber(target string) TriggerOption {
+	url, err := apis.ParseURL(target)
+	if err != nil {
+		// TODO(k15r): proper error handling here
```

Comment:
Oops

Reply:
It comes from backup and let this be here until we solve this holistically.